kv/kvserver: TestStoreScanInconsistentResolvesIntents failed

kv/kvserver.TestStoreScanInconsistentResolvesIntents [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=4369269&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=4369269&tab=artifacts#/) on master @ [bbb473c8f304ac20fec51ff0a0d04e128383bcf6](https://github.com/cockroachdb/cockroach/commits/bbb473c8f304ac20fec51ff0a0d04e128383bcf6):


```
=== RUN   TestStoreScanInconsistentResolvesIntents
    test_log_scope.go:79: test logs captured to: /artifacts/tmp/_tmp/751d67000aac5f3394c2369309253f02/logTestStoreScanInconsistentResolvesIntents2968565157
    test_log_scope.go:80: use -show-logs to present logs inline
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
Parameters in this failure:

- TAGS=bazel,gss,deadlock
</p>
</details>
/cc @cockroachdb/kv
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStoreScanInconsistentResolvesIntents.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

Easy to repro last I checked:

```
$ dev test pkg/kv/kvserver -f=TestStoreScanInconsistentResolvesIntents --stress
test_timeout=86400 --run_under '@com_github_cockroachdb_stress//:stress ' --test_filter=TestStoreScanInconsistentResolvesIntents --test_output streamed
WARNING: Ignoring JAVA_HOME, because it must point to a JDK, not a JRE.
INFO: Invocation ID: f13fc253-673c-4753-95e4-8da35fffa824
WARNING: Streamed test output requested. All tests will be run locally, without sharding, one at a time
INFO: Build options --run_under, --test_sharding_strategy, and --test_timeout have changed, discarding analysis cache.
INFO: Analyzed target //pkg/kv/kvserver:kvserver_test (567 packages loaded, 19298 targets configured).
INFO: Found 1 test target...
70 runs so far, 0 failures, over 5s
155 runs so far, 0 failures, over 10s
233 runs so far, 0 failures, over 15s
302 runs so far, 0 failures, over 20s

initialized metamorphic constant "coldata-batch-size" with value 1243
initialized metamorphic constant "kv-batch-size" with value 1
initialized metamorphic constant "datum-row-converter-batch-size" with value 1
initialized metamorphic constant "row-container-rows-per-chunk-shift" with value 1
initialized metamorphic constant "zig-zag-joiner-batch-size" with value 1
initialized metamorphic constant "spilling-queue-initial-len" with value 15
initialized metamorphic constant "merge-joiner-groups-buffer" with value 7
initialized metamorphic constant "ColIndexJoin-batch-size" with value 1201461
initialized metamorphic constant "max-batch-size" with value 8084
initialized metamorphic constant "async-IE-result-channel-buffer-size" with value 21
initialized metamorphic constant "kv.rangefeed.catchup_scan_iterator_optimization.enabled" with value false
initialized metamorphic constant "datum-row-converter-batch-size" with value 1
initialized metamorphic constant "row-container-rows-per-chunk-shift" with value 1
initialized metamorphic constant "zig-zag-joiner-batch-size" with value 1
initialized metamorphic constant "spilling-queue-initial-len" with value 2
initialized metamorphic constant "max-batch-size" with value 8773
initialized metamorphic constant "max-batch-byte-size" with value 8989082
initialized metamorphic constant "async-IE-result-channel-buffer-size" with value 18
I220113 21:18:07.508839 1 (gostd) rand.go:147  [-] 1  random seed: -3614073936900581159
panic: interface conversion: interface {} is nil, not uuid.UUID [recovered]
        panic: interface conversion: interface {} is nil, not uuid.UUID [recovered]
        panic: interface conversion: interface {} is nil, not uuid.UUID

goroutine 62 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).Recover(0xc0002d6870, {0x8fa1048, 0xc0012f90a0})
        github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:251 +0x94
panic({0x7895340, 0xc0007fcdb0})
        GOROOT/src/runtime/panic.go:1038 +0x215
github.com/cockroachdb/cockroach/pkg/kv/kvserver.(*Store).Send.func1()
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_send.go:95 +0x1ff
panic({0x7895340, 0xc0007fcdb0})
        GOROOT/src/runtime/panic.go:1047 +0x266
github.com/cockroachdb/cockroach/pkg/kv/kvserver.TestStoreScanInconsistentResolvesIntents.func1({{0x8f85358, 0xc0007fcd80}, {0xc0009923d8, 0x8}, 0x0, 0x1, {0x9014bf8, 0xc0012cc000}, {{0x16c9f126b565fa38, 0x0, ...}, ...}, ...})
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/store_test.go:2170 +0x154
github.com/cockroachdb/cockroach/pkg/kv/kvserver.evaluateBatch({_, _}, {_, _}, {_, _}, {_, _}, _, 0xc001446400, ...)
        github.com/cockroachdb/cockroach/pkg/kv/kvserver/replica_evaluate.go:255 +0x9b9
```

--- a/pkg/kv/kvserver/store_test.go
+++ b/pkg/kv/kvserver/store_test.go
@@ -2091,6 +2091,7 @@ func TestStoreScanInconsistentResolvesIntents(t *testing.T) {
 	defer log.Scope(t).Close(t)
 
 	var intercept atomic.Value
+	intercept.Store(uuid.Nil)
 	cfg := TestStoreConfig(nil)
 	cfg.TestingKnobs.EvalKnobs.TestingEvalFilter =
 		func(filterArgs kvserverbase.FilterArgs) *roachpb.Error {

